packages/app-headless-cms/src/admin/components/ContentEntries/Filters/Filters.tsx

```diff
@@ -1,11 +1,75 @@
-import React from "react";
+import React, { useCallback } from "react";
 import { Filters as BaseFilters, FiltersOnSubmit } from "@webiny/app-admin";
 import { useContentEntryListConfig } from "~/admin/config/contentEntries";
 import { useContentEntriesList } from "~/admin/views/contentEntries/hooks";
+import { AdvancedSearch, GraphQLInputMapper } from "@webiny/app-aco";
+import { useModel } from "~/admin/hooks";
+import {
+    FieldRaw,
+    QueryObjectDTO
+} from "@webiny/app-aco/components/AdvancedSearch/QueryBuilder/domain";
 
 export const Filters = () => {
     const { browser } = useContentEntryListConfig();
     const list = useContentEntriesList();
+    const { model } = useModel();
+
+    const getFields = useCallback((): FieldRaw[] => {
+        const excludedFieldTypes = ["ref", "rich-text", "file", "object", "dynamicZone"];
+
+        const defaultFields = [
+            {
+                id: "status",
+                type: "text",
+                label: "Status",
+                multipleValues: true,
+                predefinedValues: {
+                    enabled: true,
+                    values: [
+                        {
+                            label: "Draft",
+                            value: "draft"
+                        },
+                        {
+                            label: "Published",
+                            value: "published"
+                        },
+                        {
+                            label: "Unpublished",
+                            value: "unpublished"
+                        }
+                    ]
+                }
+            },
+            {
+                id: "createdOn",
+                type: "datetime",
+                label: "Created on",
+                settings: { type: "dateTimeWithoutTimezone" }
+            },
+            {
+                id: "savedOn",
+                type: "datetime",
+                label: "Modified on",
+                settings: { type: "dateTimeWithoutTimezone" }
+            }
+        ];
+
+        const fields = model.fields
+            .filter(modelField => !excludedFieldTypes.includes(modelField.type))
+            .map(modelField => {
+                return {
+                    id: modelField.fieldId,
+                    type: modelField.type,
+                    label: modelField.label,
+                    multipleValues: modelField.multipleValues || false,
+                    predefinedValues: modelField?.predefinedValues || undefined,
+                    settings: modelField.settings
+                };
+            });
+
+        return [...defaultFields, ...fields];
+    }, [model.fields]);
```

Comment:
This whole part can be extracted into:
a) a standalone hook, which lives net to the `Filters` component, just to have this outside of the component, or better...
b) create a small Presenter to handle this logic, and instantiate the Presenter with the `model` in the `useEffect` 
hook, when you have the model available

With a simple Presenter (which doesn't even have to be observable, you can use a simple callback method to get the fields) you get a better separation, and no React 😅